coingecko/src/adapter.ts

```diff
@@ -0,0 +1,42 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { price, global } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint) {
+    case price.NAME:
+      return await price.execute(request, config)
+
+    case 'globalMarketCap':
+      request.data.path = 'total_market_cap'
```

Comment:
Example of using path in the endpoint for backwards compatibility. 

If we move forward on that, `path` should get added as a param to `adapter.ts` and `global` exposed.

Reply:
I have thought of a solution similar to this -> https://github.com/smartcontractkit/external-adapters-js/blob/refactor/remove-adapter-duplication/cryptoapis/src/endpoint/bc_info.ts#L5
where different endpoint names can route to the same sub-adapter, and then the endpoint param can used for path-finding or other type of logic inside the sub-adapter. But not completed/accepted yet. What do you think?

Reply:
Opened #294